### ISSUE TITLE
Adding/editing a timezone causes a type error

### ISSUE THREAD

**Issue by critzo** [throneless-tech/murakami-viz#99]
When attempting to add edit the timezone field in the "Library", setting it to any value produces this error:
![image](https://user-images.githubusercontent.com/4633345/95109324-75a8db80-070a-11eb-90c7-b3456d66b881.png)

I would like to add something like "America/Boise"

This field also could/should be a dropdown or autocomplete field with a list of known timezones.

**Comment by jheretic** [comment]
Looked into this, and just to summarize:
* this issue affects editing libraries in general, not just timezones
* this is actually an issue with the new `library_networks` database table. In the models we delete and recreate modified rows as part of a transaction during an update, because for a `PUT` operation the standardized behavior is for it to take the entire object being updated (because a `PUT` is supposed to be idempotent) and update the object if it exists, returning in our case a `204` response. If the object does not exist, then it is supposed to create the object and return a `201`. We delete the row and recreate it to avoid a possible collision if the row exists, but this introduces a problem with foreign keys, which don't like to have the entry they're referencing deleted. Consequently, we set our join tables to have cascading deletes and updates, so that the database server handles it all transparently and internally. However, the `library_networks` table is currently not set to cascade.
* Additionally, I noticed that only the first time I submit an update does the entire library object get submitted; on subsequent saves it only sends the modified field, which then causes it to fail backend input validation (because it's expecting an entire library object). I'm not sure if this is a separate problem or has something to do with the frontend page state getting messed up by the first attempt failing. Worth checking once the above is fixed.
* Separately, we could set a timezone locale dropdown, there's probably a widget for that that'll do it automatically.

To sum up, the tasks are:

- [ ] Write a migration to add `onDelete()` and `onUpdate()` CASCADE to the `library_networks` table, see other join table migrations for the syntax.
- [ ] Subsequently test for a possible frontend state issue, and fix (make sure the whole library object is getting sent and not just the updated field)
- [ ] (Bonus) add a dropdown for valid standard timezone locales rather than just having freeform text entry. Split this out into a separate issue if decide to put it in the Future milestone.
